src/Cardano/Wallet/Primitive/Signing.hs

```diff
@@ -0,0 +1,114 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DeriveGeneric #-}
+module Cardano.Wallet.Primitive.Signing where
+
+import Prelude
+
+import Cardano.Wallet.Binary
+    ( TxWitness (..), toByteString )
+import Cardano.Wallet.Primitive.AddressDerivation
+    ( Depth (RootK), Key (..), XPrv, XPub, publicKey )
+import Cardano.Wallet.Primitive.Types
+    ( Address, Hash (..), Tx (..), TxIn, TxOut, protocolMagic )
+import Control.Monad
+    ( forM )
+import Data.ByteString
+    ( ByteString )
+import GHC.Generics
+    ( Generic )
+
+import qualified Cardano.Crypto.Wallet as CC
+import qualified Codec.CBOR.Encoding as CBOR
+
+
+
+newtype PassPhrase = PassPhrase ByteString -- TODO: Was ScrubbedBytes previously
+type TxOwnedInputs owner = [(owner, TxIn)]
+
+-- | Build a transaction
+
+-- | Construct a standard transaction
+--
+-- " Standard " here refers to the fact that we do not deal with redemption,
+-- multisignature transactions, etc.
+--
+-- TODO: re-add shuffle
+-- TODO: I removed FakeSigner/SafeSigner. Might be wrong.
+mkStdTx :: (Address -> Either e (Key 'RootK XPrv))
```

Comment:
You'll also need the passphrase to make the signature here (because the `XPrv` is actually encrypted)